frontend/catalyst/autograph.py

```diff
@@ -227,9 +222,40 @@ def g(x: float, n: int):
 
     >>> print(g(0.4, 6))
     216.4
+
+    **Example 2: as a context manager**
+
+    .. code-block:: python
+
+        def f():
+            x = 6
+            if x > 5:
+                y = x ** 2
+            else:
+                y = x ** 3
+            return y
+
+        @qjit(autograph=True)
+        def g():
+            x = 0.4
+            with disable_autograph():
+                x += f()
+            return x
+
+    >>> print(g())
+    36.4
     """
 
-    return do_not_convert(fn)
+    def __init__(self):
+        super().__init__(status=ag_ctx.Status.DISABLED)
+
+    def __enter__(self):
+        super().__enter__()
+        return self
+
+    def __exit__(self, unused_type, unused_value, unused_traceback):
+        super().__exit__(unused_type, unused_value, unused_traceback)
+        return False
```

Comment:
I don't think we need to redefine them if we only call the super method anyways :)

Reply:
True, only `__init__` with disabled status is needed, is that correct? If yes, then it's fixed now